app/src/main/java/com/bogdan/codeforceswatcher/features/users/UserActivity.kt

```diff
@@ -58,38 +61,53 @@ class UserActivity : AppCompatActivity() {
 
     private fun displayUser() {
         tvRank.text = user.buildRank()
-        tvCurrentRating.text = user.buildRating()
-        tvUserHandle.text = getString(R.string.name, user.buildName())
-        tvMaxRating.text = user.buildMaxRating()
+        tvUserRating.text = user.buildRating()
+        tvUserHandle.text = user.buildName()
         (ivUserAvatar as CircleImageView).borderColor = ContextCompat.getColor(this, getColorByUserRank(user.rank))
 
         Picasso.get().load(avatar(user.avatar)).into(ivUserAvatar)
-        title = user.handle
+        title = user.buildScreenTitle()
     }
 
-    private fun User.buildRank() = if (rank == null) {
-        getString(R.string.rank, getString(R.string.none))
-    } else {
-        getString(R.string.rank, rank)
-    }
+    private fun User.buildRank() = rank?.let { colorTextByUserRank(it.capitalize(), it) }
+            ?: getString(R.string.none)
+
+    private fun User.buildRating(): SpannableString {
+        val str = SpannableString(
+                getString(
+                        R.string.rating,
+                        rating?.toString() ?: getString(R.string.none),
+                        maxRating?.toString() ?: getString(R.string.none)
+                )
+        )
+
+        rating?.let {
+            val startIndex = str.indexOf(it.toString())
+            val color = getColorByUserRank(rank)
+            str.colorSubstring(startIndex, startIndex + it.toString().length, color)
+        }
+
+        maxRating?.let {
+            val startIndex = str.lastIndexOf(it.toString())
+            val color = getColorByUserRank(maxRank)
+            str.colorSubstring(startIndex, startIndex + it.toString().length, color)
+        }
 
-    private fun User.buildRating() = if (rating == null) {
-        getString(R.string.cur_rating, getString(R.string.none))
-    } else {
-        getString(R.string.cur_rating, rating.toString())
+        return str
     }
 
-    private fun User.buildName() = when {
+    private fun User.buildName() = colorTextByUserRank(when {
         firstName == null && lastName == null -> getString(R.string.none)
-        firstName == null -> lastName
-        lastName == null -> firstName
+        firstName == null -> lastName!!
```

Comment:
`?.orEmpty()` or `?: ""` and the same for all new `!!`. Let's not risk crashing the app.